src/Plugin/Field/FieldFormatter/EmbeddableFormatter.php

```diff
@@ -55,17 +54,34 @@ public static function isApplicable(FieldDefinitionInterface $field_definition)
       return FALSE;
     }
 
-    $media_type = MediaType::load($field_definition->getTargetBundle());
+    $media_type = self::getMediaType($field_definition->getTargetBundle());
     return $media_type && $media_type->getSource() instanceof Embeddable;
   }
 
+  /**
+   * Load the media type from the media type id.
+   *
+   * @param string $media_type_id
+   *   Media type machine name.
+   *
+   * @return \Drupal\media\MediaTypeInterface|null
+   *   Media type entity.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  protected static function getMediaType($media_type_id) {
+    return \Drupal::entityTypeManager()
```

Comment:
DI

Reply:
static method. can't use DI

Reply:
its being used by another static method so thats why.

Reply:
Oh stupid static.

Reply:
Protected static tho?

Reply:
yeah. it's only used "internally". I've seen it as an approach in Drupal core for some scenarios